File: C/Eventplanning.c
#include <stdio.h>
#include <limits.h>

int main() {
    int people, budget, hotels, weekends;
    scanf("%d %d %d %d", &people, &budget, &hotels, &weekends);

    int cheapest = INT_MAX;
    for (int i = 0; i < hotels; i++) {
        int price;
        scanf("%d", &price);
        for (int j = 0; j < weekends; j++) {
            int rooms;
            scanf("%d", &rooms);
            if (rooms >= people) {
                int checker = people * price;
                if (checker <= budget && checker < cheapest) {
                    cheapest = checker;
                }
            }
        }
    }
    if (cheapest < INT_MAX)
        printf("%d", cheapest);
    else
        printf("stay home");
}

File: Java/Eventplanning.java
import java.util.Scanner;

class Eventplanning {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int people = sc.nextInt();
        int budget = sc.nextInt();
        int hotels = sc.nextInt();
        int weekends = sc.nextInt();
        int cheapest = Integer.MAX_VALUE;

        for (int i = 0; i < hotels; i++) {
            int price = sc.nextInt();
            for (int j = 0; j < weekends; j++) {
                int rooms = sc.nextInt();
                if (rooms >= people) {
                    int checker = people * price;
                    if (checker <= budget && checker < cheapest) {
                        cheapest = checker;
                    }
                }
            }
        }
        System.out.println(cheapest < Integer.MAX_VALUE ? cheapest : "stay home");
        sc.close();
    }
}